tooling/noir_js/src/witness_generation.ts

```diff
@@ -1,19 +1,21 @@
 import { abiEncode, InputMap } from '@noir-lang/noirc_abi';
 import { base64Decode } from './base64_decode.js';
-import { executeCircuit, WitnessMap } from '@noir-lang/acvm_js';
+import { executeCircuit, WitnessMap, ForeignCallHandler, ForeignCallInput } from '@noir-lang/acvm_js';
 import { CompiledCircuit } from '@noir-lang/types';
 
+const defaultForeignCallHandler: ForeignCallHandler = (name: string, args: ForeignCallInput[]) => {
```

Comment:
Is name the name of the foreign function? It so, maybe later on we can put together some handlers we can match with this

Reply:
Yep although I'm not wanting to do this in this PR

> I've neglected providing a default implementation for println as we'd need to replicate the string formatting logic which exists in the Rust code. For now we can just direct users to provide their own foreign call handler and make them responsible for printing results as they like.

Reply:
can we open an issue for this @TomAFrench ?